TaylorSource/CoreData/NSFRCUpdateHandler.swift

```diff
@@ -0,0 +1,115 @@
+import UIKit
+import CoreData
+
+private class WeakObserver {
+    private(set) weak var value: NSFRCIndexedUpdateConsumer?
+
+    init(_ observer: NSFRCIndexedUpdateConsumer) {
+        value = observer
+    }
+}
```

Comment:
I think for a class named `WeakObserver` then it should either be generic:

```swift
private class WeakObserver<T: AnyObject> {
    private(set) weak var value: T?

     init(_ observer: T) {
        value = observer
    }
}
```

and used like this:

```swift
private var observers = [WeakObserver< NSFRCIndexedUpdateConsumer>]()
```

Alternatively, could rename it to something less general, maybe `NSFRCIndexedUpdateConsumerWeakContainer` 😒, or nest the type as it is inside the `NSFRCUpdateHandler`.

This way, we're not populating the module namespace with general sounding classes which are actually quite bespoke.

I think nesting the type is the most pragmatic option for now. Maybe eventually it'll get pulled out and made truly generic.